In [1]:
import pandas as pd
df = pd.read_csv('titanic.csv')

In [2]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [3]:
#Análisis de superviviencia
conteo=df['Survived'].value_counts()
total=len(df)
porcentaje=(conteo/total)*100
print(porcentaje.round(2))

Survived
0    61.62
1    38.38
Name: count, dtype: float64


In [4]:
#Análisis de clase
conteo = df['Pclass'].value_counts()
total = len(df)
porcentaje = (conteo/total)*100
print(porcentaje.round(2))

Pclass
3    55.11
1    24.24
2    20.65
Name: count, dtype: float64


In [5]:
# Análisis por género
df['Sex'].value_counts()


,count
Sex,
male,577
female,314


#Conclusiones
A partir del análisis del dataset del Titanic, se identificaron varios aspectos relevantes. El porcentaje de supervivencia fue del 38.38%, lo que indica que la mayoría de los pasajeros no logró sobrevivir al accidente. Este resultado sugiere que se trató de un evento de gran magnitud, en el que las condiciones de respuesta ante la emergencia pudieron no haber sido suficientes.

En cuanto a la clase socioeconómica, se observó que el 55.10% de los pasajeros pertenecía a la tercera clase. Esto indica que una gran parte de los viajeros correspondía a personas de recursos limitados. Considerando el contexto histórico, es posible que muchos de estos pasajeros fueran migrantes que se dirigían a Estados Unidos en busca de mejores oportunidades.

Respecto al género, se evidenció que había una mayor cantidad de hombres que de mujeres a bordo. Este comportamiento podría estar relacionado con dinámicas sociales de la época.